test/Microsoft.TestPlatform.Common.UnitTests/ExtensionFramework/Utilities/TestExtensionsTests.cs

```diff
@@ -239,4 +241,110 @@ public void GetExtensionsDiscoveredFromAssemblyShouldReturnTestDiscoveresAndLogg
+        Assert.IsFalse(ReferenceEquals(merged1, first));
+        Assert.IsFalse(ReferenceEquals(merged1, second));
+        Assert.IsFalse(ReferenceEquals(merged2, first));
+        Assert.IsFalse(ReferenceEquals(merged2, second));
+        Assert.IsFalse(ReferenceEquals(merged1, merged2));
+    }
+
+    [TestMethod]
+    public void MergedDictionaryShouldBeSuccessful()
+    {
+        var first = new Dictionary<string, HashSet<string>>
+        {
+            // Merged with "key1" from the next set.
+            { "key1", new HashSet<string>(new List<string>() { "ext1", "ext2", "ext3" }) },
+            // Empty hashset, will be removed from the result.
+            { "key2", new HashSet<string>() },
+            // Added as is.
+            { "key5", new HashSet<string>(new List<string>() { "ext1", "ext2" }) }
+        };
+        var second = new Dictionary<string, HashSet<string>>
+        {
+            // Merged with "key1" from the previous set.
+            { "key1", new HashSet<string>(new List<string>() { "ext2", "ext3", "ext3", "ext4", "ext5" }) },
+            // Empty hashset, will be removed from the result.
+            { "key2", new HashSet<string>() },
+            // Empty hashset, will be removed from the result.
+            { "key3", new HashSet<string>() },
+            // Added as is.
+            { "key4", new HashSet<string>(new List<string>() { "ext1" }) }
+        };
+        var expected = new Dictionary<string, HashSet<string>>
+        {
+            { "key1", new HashSet<string>(new List<string>() { "ext1", "ext2", "ext3", "ext4", "ext5" }) },
+            { "key4", new HashSet<string>(new List<string>() { "ext1" }) },
+            { "key5", new HashSet<string>(new List<string>() { "ext1", "ext2" }) }
+        };
+
+        // Merge the two dictionaries.
+        var merged = TestExtensions.CreateMergedDictionary(first, second);
+
+        // Make sure the merged dictionary has the exact same keys as the expected dictionary.
+        Assert.IsTrue(merged.Count == expected.Count);
+        Assert.IsFalse(merged.Where(kvp => !expected.ContainsKey(kvp.Key)).Any());
+        Assert.IsFalse(expected.Where(kvp => !merged.ContainsKey(kvp.Key)).Any());
+
+        // Make sure the hashsets for each key are equal.
+        Assert.IsFalse(merged.Where(kvp => !kvp.Value.SequenceEqual(expected[kvp.Key])).Any());
+    }
+
+    [TestMethod]
+    public void AddExtensionTelemetryShouldBeSuccessful()
+    {
+        var telemetryData = new Dictionary<string, object>();
+        var extensions = new Dictionary<string, HashSet<string>>
+        {
+            { "key1", new HashSet<string>(new List<string>() { "ext1", "ext2", "ext3", "ext4", "ext5" }) },
+            { "key4", new HashSet<string>(new List<string>() { "ext1" }) },
+            { "key5", new HashSet<string>(new List<string>() { "ext1", "ext2" }) }
+        };
+
+        var expectedTelemetry =
+            "{\"key1\":[\"ext1\",\"ext2\",\"ext3\",\"ext4\",\"ext5\"],"
```

Comment:
I think we should be able to use the raw string feature which would increase readability here.